int overflow

Greetings!

```sh
$ go19 build -ldflags=-s .
# github.com/ericlagergren/ocb3
C:\Users\0wner\go\pkg\mod\github.com\ericlagergren\ocb3@v0.0.0-20220206064240-ec5c2d9fa949\ocb3.go:258:40: maxInputSize (untyped int constant 4503599627370496) overflows int
C:\Users\0wner\go\pkg\mod\github.com\ericlagergren\ocb3@v0.0.0-20220206064240-ec5c2d9fa949\ocb3.go:354:41: maxInputSize (untyped int constant 4503599627370496) overflows int
```

How to proceed?

Thanks in advance.

--- a/ocb3.go
+++ b/ocb3.go
@@ -254,8 +254,8 @@ func (a *aead) Seal(dst, nonce, plaintext, additionalData []byte) []byte {
 	if a.tagSize < minTagSize {
 		panic("ocb3: invalid tag length")
 	}
-	if len(plaintext)+len(additionalData) < 0 ||
-		len(plaintext)+len(additionalData) > maxInputSize {
+	if uint64(len(plaintext)) > maxInputSize ||
+		maxInputSize-uint64(len(plaintext)) < uint64(len(additionalData)) {
 		panic("ocb3: message too large")
 	}
 
@@ -350,8 +350,8 @@ func (a *aead) Open(dst, nonce, ciphertext, additionalData []byte) ([]byte, erro
 	if len(ciphertext) < a.tagSize {
 		return nil, errOpen
 	}
-	if len(ciphertext)+len(additionalData) < 0 ||
-		len(ciphertext)+len(additionalData) > maxInputSize+a.tagSize {
+	if uint64(len(ciphertext)) > maxInputSize ||
+		(maxInputSize+uint64(a.tagSize))-uint64(len(ciphertext)) < uint64(len(additionalData)) {
 		panic("ocb3: message too large")
 	}
 

